impeller/renderer/compute_pass.h

```diff
@@ -35,6 +35,25 @@ class ComputePass : public ResourceBinder {
 
   virtual fml::Status Compute(const ISize& grid_size) = 0;
 
+  /// @brief Ensures all previously encoded commands have finished writing to
+  ///        buffers before any following commands can read from those buffer.
+  ///
+  ///        This method after encoding a command that writes to a storage
+  ///        buffer that a subsequent command will read from, provided that
+  ///        is a compute command and not a render pass command. It does not
+  ///        matter if the compute command is in a different command buffer.
+  virtual void AddBufferMemoryBarrier() = 0;
+
+  /// @brief Ensures all previously encoded commands have finished writing to
+  ///        textures before any following commands can read from those
+  ///        textures.
+  ///
+  ///        This method after encoding a command that writes to a storage
```

Comment:
This sentence is quite confusing to me. Did you mean "Call this method?" as a recommended usage?

Reply:
Yeah removed, I don't know what I was doing here